source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -206,6 +206,11 @@ FilterConfig::FilterConfig(
         fmt::format("OAuth2 filter: invalid authorization endpoint URL '{}' in config.",
                     authorization_endpoint_));
   }
+
+  if (proto_config.has_retry_policy()) {
+    retry_policy_ = Http::Utility::convertCoreToRouteRetryPolicy(
+        proto_config.retry_policy(), "5xx,gateway-error,connect-failure,reset");
+  }
```

Comment:
Set a type alias name to simplify the code:
like:
```
using RouteRetryPolicy = envoy::config::route::v3::RetryPolicy;
```

And make the retry_policy_ optional (?) by using `absl::optional<RouteRetryPolicy>`